CONTRIBUTING.md

```diff
@@ -26,6 +27,132 @@ The easier it is for us to reproduce it, the faster it'll be fixed!
+    libselinux-devel \
+    make \
+    ostree-devel \
+    pkgconfig \
+    runc \
+    skopeo-containers
+```
+
+For [other distos you can find more available commands](install.md) on the install documentation to prepare your environment.
+
+Check that the installed version of `runc` is new enough by using:
+```shell
+$ runc --version
+runc --version
+runc version 1.0.0-rc6+dev
+commit: d164d9b08bf7fc96a931403507dd16bced11b865
+spec: 1.0.1-dev
+```
+
+You also need to check the installed version of golang, 1.10.x or higher is required:
+```shell
+$ go version
+go version go1.10.3 gccgo (GCC) 8.2.1 20181215 (Red Hat 8.2.1-6) linux/amd64
+```
+
+### Fork and clone libpod
+
+First you need to fork this project on github.
+
+Clone this project locally:
+```shell
+$ go get github.com/<you>/libpod
+```
+
+### Deal with make
+
+Libpod use a Makefile to realize common action like building etc...
+
+You can list available actions by using:
+```shell
+$ make help
+Usage: make <target>
+ * 'binaries' - Build podman
+ * 'changelog' - Generate changelog
+ * 'clean' - Clean artifacts
+ * 'docs' - Generate documentation
+ * 'gofmt' - Verify the source code gofmt
+ * 'install' - Install binaries to system locations
+ * 'integration' - Execute integration tests
+ * 'libpodimage' - Build the libpod image
+ * 'lint' - Execute the source code linter
+ * 'local-cross' - Cross local compilation
+ * 'perftest' - Build perf tests
+ * 'podman' - Build with podman
+ * 'podman-remote' - Build with podman on remote environment
+ * 'podman-remote-darwin' - Build with podman on remote OSX environment
+ * 'run-perftest' - Build and run perf tests
+ * 'shell' - Run the built image and attach a shell
+ * 'test' - Run tests on built image
+ * 'testunit' - Run unittest on the built image
+ * 'varlink_generate' - Generate varlink
+...
+```
+
+### Install tools
+
+Makefile allow you to install needed tools:
+```shell
+$ make install.tools
+```
+
+### Building binaries and test your changes
+
+When you start on hack on you surely want to test your changes then `make binaries` do this.
+
+Your binaries was created inside the `bin/` directory and you can test your changes:
+```shell
+$ bin/podman -h
+bin/podman -h
+NAME:
+   podman - manage pods and images
+
+USAGE:
+   podman [global options] command [command options] [arguments...]
+
+VERSION:
+   1.0.1-dev
+
+COMMANDS:
+     attach           Attach to a running container
+     build            Build an image using instructions from Dockerfiles
+     commit           Create new image based on the changed container
+     container        Manage Containers
+     cp               Copy files/folders between a container and the local filesystem
+```
+
+Well! you can now create your own branch and hack on it and then submitting your pull request.
+
+For further reading about branching [you can read this document](https://herve.beraud.io/containers/linux/podman/isolate/environment/2019/02/06/how-to-hack-on-podman.html).
+
```

Comment:
how about a note to be sure to add an integration test for your changes so we can protect against regressions?

Reply:
Now idea how to prevent it... I doesn't plan to remove it, but if is an issue for you I can remove it.